cmd/frontend/graphqlbackend/default_settings.go

```diff
@@ -0,0 +1,62 @@
+package graphqlbackend
+
+import (
+	"context"
+	"encoding/json"
+
+	graphql "github.com/graph-gophers/graphql-go"
+	"github.com/graph-gophers/graphql-go/relay"
+	"github.com/sourcegraph/sourcegraph/pkg/api"
+)
+
+var builtinExtensionIDs = []string{
+	"sourcegraph/typescript",
+	"sourcegraph/python",
+	"sourcegraph/java",
+	"sourcegraph/go",
+	"sourcegraph/cpp",
+	"sourcegraph/ruby",
+	"sourcegraph/php",
+	"sourcegraph/csharp",
+	"sourcegraph/shell",
+	"sourcegraph/scala",
+}
+
+const singletonDefaultSettingsGQLID = "Default settings"
+
+type defaultSettingsResolver struct {
+	gqlID string
+}
+
+var singletonDefaultSettingsResolver = &defaultSettingsResolver{gqlID: singletonDefaultSettingsGQLID}
+
+func marshalDefaultSettingsGQLID(defaultSettingsID string) graphql.ID {
+	return relay.MarshalID("Default settings", defaultSettingsID)
```

Comment:
Generally caps case with no space

https://sourcegraph.sgdev.org/search?q=relay.MarshalID

```suggestion
	return relay.MarshalID("DefaultSettings", defaultSettingsID)
```

I would probably just reuse `singletonDefaultSettingsGQLID` here too.